packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -3053,3 +3119,135 @@ class _WhitespaceDirectionalityFormatter extends TextInputFormatter {
     return _ltrRegExp.hasMatch(String.fromCharCode(value)) ? TextDirection.ltr : TextDirection.rtl;
   }
 }
+
+class _ScribbleElement extends StatefulWidget {
+  const _ScribbleElement({Key? key, required this.child, required this.focusNode, required this.editableKey}): super(key: key);
+
+  final Widget child;
+  final FocusNode focusNode;
+  final GlobalKey editableKey;
+
+  @override
+  _ScribbleElementState createState() => _ScribbleElementState();
+}
+
+class _ScribbleElementState extends State<_ScribbleElement> implements ScribbleClient {
+  _ScribbleElementState(): _elementIdentifier = (_nextElementIdentifier++).toString();
+
+  @override
+  void initState() {
+    super.initState();
+    TextInput.registerScribbleElement(elementIdentifier, this);
+  }
+
+  @override
+  void dispose() {
+    super.dispose();
+    TextInput.unregisterScribbleElement(elementIdentifier);
+  }
+
+  RenderEditable? get renderEditable => widget.editableKey.currentContext?.findRenderObject() as RenderEditable?;
+
+  static int _nextElementIdentifier = 1;
+  final String _elementIdentifier;
+
+  @override
+  String get elementIdentifier => _elementIdentifier;
+
+  @override
+  void onScribbleFocus(Offset offset) {
+    widget.focusNode.requestFocus();
+    renderEditable?.selectPositionAt(from: offset, cause: SelectionChangedCause.keyboard);
+  }
+
+  @override
+  bool isInScribbleRect(Rect rect) {
+    final Rect _bounds = bounds;
+    if (renderEditable?.readOnly ?? false)
+      return false;
+    if (_bounds == Rect.zero)
+      return false;
+    if (!_bounds.overlaps(rect))
+      return false;
+    final Rect intersection = _bounds.intersect(rect);
+    return <Offset>[
+      intersection.topLeft,
+      intersection.topCenter,
+      intersection.topRight,
+      intersection.centerLeft,
+      intersection.center,
+      intersection.centerRight,
+      intersection.bottomLeft,
+      intersection.bottomCenter,
+      intersection.bottomRight
+    ].any((Offset point) {
+      final HitTestResult result = HitTestResult();
+      WidgetsBinding.instance?.hitTest(result, point);
+      return result.path.any((HitTestEntry entry) => entry.target == renderEditable);
+    });
+  }
+
+  @override
+  Rect get bounds {
+    final RenderBox? box = context.findRenderObject() as RenderBox?;
+    if (box == null || !mounted || !box.attached)
+      return Rect.zero;
+    final Offset topLeft = box.localToGlobal(Offset.zero);
```

Comment:
This may give you the wrong size because only the origin is transformed. The transform can be retrieved using `getTransformTo` and it should be applied to the rect. There doesn't seem to a shorthand method for this unfortunately.